tests/test_api.py

```diff
@@ -33,18 +32,29 @@ def setUp(self):
         self.username = "journalist"
         self.password = "correct horse battery staple profanity oil chewy"
         self.server = "http://127.0.0.1:8081/"
-        self.api = API(self.server, self.username, self.password, str(self.totp.now()))
+
+        # Because we may be using a TOTP code from a previous run that has since
+        # been invalidated (or that may be invalid because of bad timing),
+        # we try repeatedly to get the token with a new TOTP code.
+        #
+        # Once we have a token, it'll be valid for several hours, so we stash
+        # it away for future use by other tests.
         for i in range(3):
+            totp = self.totp.now()
+            self.api = API(self.server, self.username, self.password, str(totp))
             try:
                 self.api.authenticate()
-            except BaseError:
+            except AuthError:
```

Comment:
AuthError is the more specific, occasionally expected error, so it seemed the more appropriate one to catch here so we don't accidentally swallow other useful exceptions.